core/src/main/java/org/eclipse/hono/config/ClientConfigProperties.java

```diff
@@ -575,4 +603,140 @@ public final Pattern getAddressRewritePattern() {
     public final String getAddressRewriteReplacement() {
         return addressRewriteReplacement;
     }
+
+    /**
+     * Gets the minimum size of AMQP 1.0 messages that this client requires a peer to accept.
+     * <p>
+     * The default value of this property is {@value #MIN_MESSAGE_SIZE_NONE}.
+     * <p>
+     * The value of this property will be used by the client during sender link establishment.
+     * Link establishment will fail, if the peer indicates a max-message-size in its <em>attach</em>
+     * frame that is smaller than the configured minimum message size.
+     *
+     * @return The message size in bytes or {@value #MIN_MESSAGE_SIZE_NONE}, indicating no minimum size at all.
+     */
+    public final long getMinMessageSize() {
+        return minMessageSize;
```

Comment:
I think this property is quite confusing. Just guessing from the name, I would think that with such a setting, messages with a smaller size than configured here would be dropped/rejected. However, its actual semantics would be described by a name like `getMinimumRequiredPeerMaxMessageSize()` (which is maybe a bit long), right?
Also the fact that `getMaxMessageSize` only applies to a receiver whereas `getMinMessageSize` is only for a sender makes things somewhat hard to grasp at first sight.

Can there be a better name? Or, do we really need this property?

Reply:
> Or, do we really need this property?

What is the alternative? Having the sender link being closed on each message being sent that is bigger than the peer's max-message-size?

Reply:
> What is the alternative? Having the sender link being closed on each message being sent that is bigger than the peer's max-message-size?

Right, having that check on link establishment already is better. For the use case of the telemetry/event sender clients in the protocol adapters, I think this config value should internally be modified to be at least the configured `ServiceConfigProperties#getMaxPayloadSize` value.

I still think the name should be changed, though. How about `getMinMaxMessageSize` or `getMinMessageSizeLimit`?
(MinMaxMessageSize would be similar in naming to the MIN-MAX-FRAME-SIZE constant of the AMQP spec).

Reply:
let's use *minMaxMessageSize* then ...